src/test/java/mate/academy/dao/UserDaoTest.java

```diff
@@ -0,0 +1,75 @@
+package mate.academy.dao;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.impl.RoleDaoImpl;
+import mate.academy.dao.impl.UserDaoImpl;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import org.hibernate.SessionFactory;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UserDaoTest extends AbstractTest {
+    private User userAlice;
+    private User userBob;
+    private UserDao userDao;
+    private RoleDao roleDao;
+    
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[]{User.class, Role.class};
+    }
+    
+    @BeforeEach
+    void setUp() {
+        SessionFactory sessionFactory = getSessionFactory();
+        roleDao = new RoleDaoImpl(sessionFactory);
+        Role adminRole = new Role(Role.RoleName.ADMIN);
+        roleDao.save(adminRole);
+        userAlice = new User();
+        userAlice.setEmail("alice@gmail.com");
+        userAlice.setPassword("qwerty");
+        userAlice.setRoles(Set.of(adminRole));
+        userBob = new User();
+        userBob.setEmail("bob@gmail.com");
+        userBob.setPassword("1234");
+        userDao = new UserDaoImpl(sessionFactory);
+        userDao.save(userAlice);
+    }
+    
+    @Test
+    void save_Ok() {

```

Comment:
```suggestion
    void save_ok() {
```
the same for all methods in all classes